common-libs/user-management/src/TrueCallerPublicKeyManager.ts

```diff
@@ -0,0 +1,43 @@
+import { fetch } from '@affinidi/platform-fetch'
+import SdkErrorFromCode from './SdkErrorFromCode'
+
+const TRUE_CALLER_PUBLIC_KEY_URL = 'https://api4.truecaller.com/v1/key'
+
+export class TrueCallerPublicKeyManager {
+  constructor(
+    private readonly trueCallerPublicKeyUrl: string,
+    private keys: Array<{ keyType: string; key: string }> = [],
+  ) {}
+
+  /**
+   //  * Fetch `Truecaller` public key.
+   //  * See docs https://docs.truecaller.com/truecaller-sdk/android/server-side-response-validation/for-truecaller-users-verification-flow
+   //  */
+  async sync() {
+    let result: Array<{ keyType: string; key: string }>
+    try {
+      const resp = await fetch(this.trueCallerPublicKeyUrl, {
+        method: 'GET',
+        headers: {
+          Accept: 'application/ld+json, application/json',
+        },
+      })
+
+      result = await resp.json()
+    } catch (error) {
+      throw new SdkErrorFromCode('UM-5', { error })
+    }
+
+    if (result.length < 1) {
+      throw new SdkErrorFromCode('UM-4')
+    } else {
+      this.keys = result
```

Comment:
```suggestion
      this.trueCallerPK = result[0] as TrueCallerPK
```
and then below just return trueCallerPK